src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -191,84 +191,88 @@ public ITextBufferAssociatedViewService AssociatedViewService
             }
 #endif
 
+            var updatedSolution = operations.OfType<ApplyChangesOperation>().FirstOrDefault()?.ChangedSolution ?? oldSolution;
             TryStartRenameSession(workspace, oldSolution, updatedSolution, cancellationToken);
         }
 
-        private static async Task<Solution> ProcessOperationsAsync(
-            Workspace workspace, Document fromDocument, string title, Solution oldSolution, Solution updatedSolution, List<CodeActionOperation> operationsList, 
-            IProgressTracker progressTracker, CancellationToken cancellationToken)
+        private TextDocument TryGetSingleChangedText(
+            Solution oldSolution, ImmutableArray<CodeActionOperation> operationsList)
         {
-            foreach (var operation in operationsList)
+            Debug.Assert(operationsList.Length > 0);
+            if (operationsList.Length > 1)
             {
-                var applyChanges = operation as ApplyChangesOperation;
-                if (applyChanges == null)
-                {
-                    operation.Apply(workspace, cancellationToken);
-                    continue;
-                }
+                return null;
+            }
 
-                // there must be only one ApplyChangesOperation, we will ignore all other ones.
-                if (updatedSolution == oldSolution)
-                {
-                    updatedSolution = applyChanges.ChangedSolution;
-                    var projectChanges = updatedSolution.GetChanges(oldSolution).GetProjectChanges();
-                    var changedDocuments = projectChanges.SelectMany(pd => pd.GetChangedDocuments());
-                    var changedAdditionalDocuments = projectChanges.SelectMany(pd => pd.GetChangedAdditionalDocuments());
-                    var changedFiles = changedDocuments.Concat(changedAdditionalDocuments).ToList();
-
-                    // 0 file changes
-                    if (changedFiles.Count == 0)
-                    {
-                        operation.Apply(workspace, cancellationToken);
-                        continue;
-                    }
+            var applyOperation = operationsList.Single() as ApplyChangesOperation;
+            if (applyOperation == null)
+            {
+                return null;
+            }
```

Comment:
It seems uber-paranoid to test `Length` and then use `Single()`. Couldn't you just have the following and remove the code that checks the `Length > 1` above?

```C#
var applyOperation = operationList.SingleOrDefault() as ApplyChangesOperation;
if (applyOperation == null)
{
    return null;
}
```